### ISSUE TITLE
Reproducibility

### ISSUE THREAD

**Issue by VishnuBhaarath** [dmlc/xgboost#7458]
How to get reproducible result, is it possible to get the exact same result each time, I am getting different results each time I run the model with the same set of hyperparameters.

**Comment by trivialfis** [comment]
Could you please share the script that's not reproducible?  For most of the single node training cases, it should be bit by bit reproducible given the same environment (seed, hardware, threads etc).

**Comment by trivialfis** [comment]
Also, it should be bit by bit reproducible for dask interface in distributed environment, given that the input is exactly the same (same partitioning, order, workers etc)

**Comment by trivialfis** [comment]
So, would be great if you can share a sample that's not reproducible and we can debug it.

**Comment by VishnuBhaarath** [comment]
Thanks @trivialfis I was able to rectify it, the model is indeed reproducible given the same set of hyperparameters and the same environment, I was making a mistake in partitioning of the data that's the reason for different results.
